source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/typeaheadsearch/TypeAheadSearchViewModel.kt

```diff
@@ -0,0 +1,200 @@
+package io.adaptivecards.renderer.typeaheadsearch
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.recyclerview.widget.RecyclerView
+import io.adaptivecards.R
+import io.adaptivecards.objectmodel.ChoiceInput
+import io.adaptivecards.renderer.SingleLiveEvent
+import io.adaptivecards.renderer.http.HttpRequestResult
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class TypeAheadSearchViewModel : ViewModel() {
+
+    private var staticChoicesCount: Int
+    private var titleList: List<String>
+    private var valueList: List<String>
+
+    private lateinit var dataset: String
+    private lateinit var dataType: String
+    private lateinit var adapter : FilteredAdapter
+
+    fun getAdaptor(): FilteredAdapter {
+        return adapter
+    }
+
+    private val _queryText: SingleLiveEvent<String> by lazy {
+        SingleLiveEvent()
+    }
+
+    private val _uiState: SingleLiveEvent<DynamicTypeAheadUiState> by lazy {
+        SingleLiveEvent()
+    }
+
+    val queryText: SingleLiveEvent<String> get() = _queryText
+
+    val uiState: SingleLiveEvent<DynamicTypeAheadUiState> get() = _uiState
+
+    init {
+        titleList = ArrayList()
+        valueList = ArrayList()
+        staticChoicesCount = 0
+        _queryText.value = ""
+
+        // Show start searching view by default
+        _uiState.postValue(DynamicTypeAheadUiState.SearchOptions)
+    }
+
+    fun init(titles: List<String>, values: List<String>, type: String, set: String) {
+        // get navigation params and initiate choices.data, static choices etc.
+        titleList = titles
+        valueList = values
+        staticChoicesCount = titleList.size
+
+        dataType = type
+        dataset = set
+
+        adapter = FilteredAdapter(titleList, valueList)
+        showDefaultView()
+    }
+
+    private fun showDefaultView() {
+        if (staticChoicesCount == 0) {
+            _uiState.postValue(DynamicTypeAheadUiState.SearchOptions)
+        }
+        else {
+            // Show static choices when present
+            _uiState.postValue(DynamicTypeAheadUiState.ShowingChoices)
+        }
+    }
+
+
+    fun fetchDynamicOptions(queryText: String) {
+        if (queryText.isEmpty()) {
+            clearText()
+            return
+        }
+
+        _queryText.postValue(queryText)
+        _uiState.postValue(DynamicTypeAheadUiState.Loading)
+
+        viewModelScope.launch(Dispatchers.IO) {
+            var result: HttpRequestResult<List<ChoiceInput>>? = null
+            DynamicTypeAheadService.getChoicesResolver()?.let {
+                // TODO : Get count from host config
+                result = it.getDynamicChoices(dataType, dataset, queryText, 15, 0)
```

Comment:
in order to ease the upcoming change, can you declare a const variable that tracks this number (15), it's at least referenced in three spots (line 95, 102, 103)

Reply:
updated